### PR TITLE
Add error log for unsupported keys in plugin config

### PR DESCRIPTION
**Pull request by michalkulakowski**
JIRA:CVS-81277

### CODE REVIEW

**Comment by atobiszei** [comment]
> _ No description provided. _

Please add issue id.

File: src/modelmanager.cpp [hunk 1]
```diff
@@ -135,6 +135,17 @@ void ModelManager::logPluginConfiguration() {
...
+        allSupportedConfigKeys.insert(
+            allSupportedConfigKeys.end(),
+            std::make_move_iterator(pluginSupportedConfigKeys.begin()),
+            std::make_move_iterator(pluginSupportedConfigKeys.end())
+        );
+    }
+    for (auto& config : pluginConfig) {
+        if (std::find(allSupportedConfigKeys.begin(), allSupportedConfigKeys.end(), config.first) == allSupportedConfigKeys.end())
+        {
+            SPDLOG_LOGGER_WARN(modelmanager_logger, "Plugin config key: {} not found in supported config keys for available devices", config.first);
```
Comment (atobiszei) [inline src/modelmanager.cpp:147]: Should we just stop at logging warning?
User specified that model should load with specified plugin configuration. Model cannot load with that configuration so we skip it and just print warning?

Either way I am not sure if we shouldn't put similar logic inside model Instance?

File: src/plugin_configuration.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "plugin_configuration.hpp"
+
+#include "logging.hpp"
+
+namespace ovms {
+Status validatePluginConfiguration(const plugin_config_t& pluginConfig, const std::string& targetDevice, const ov::Core& ieCore) {
```
Comment (dkalinowski) [inline src/plugin_configuration.cpp:22]: Isnt this related to openvino utils? We may place it in ov_utils.hpp. What do you think?

File: src/modelinstance.cpp [hunk 3]
```diff
@@ -685,6 +686,12 @@ Status ModelInstance::loadOVCompiledModel(const ModelConfig& config) {
             config.getTargetDevice());
         return status;
     }
+    auto status = validatePluginConfiguration(config.getPluginConfig(), config.getTargetDevice(), ieCore);
```
Comment (dkalinowski) [inline src/modelinstance.cpp:689]: Why do we validate configuration twice? Once after config parse and here after model compilation?

**Review by atobiszei — APPROVED** [review]

**Review by dkalinowski — CHANGES_REQUESTED** [review]
Could we add unit test checking if passing CPU_THROUGHPUT_STREAMS works but WRONG_KEY does not?

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**